### PR TITLE
Adjust namespaces and add missing XML doc comments

### PR DESCRIPTION
**Pull request by MarcinHoppe**
Relates to Particular/PlatformDevelopment#730

### CODE REVIEW

**Comment by MarcinHoppe** [comment]
@Particular/tf-azure-host-beta @Particular/azure-host-maintainers Please review and merge.

The shared host sample will be updated in a separate PR.

File: src/NServiceBus.Hosting.Azure/Roles/IConfigureThisHost.cs [hunk 1]
```diff
@@ -1,9 +1,14 @@
 namespace NServiceBus
 {
-    using Hosting;
-
+    /// <summary>
+    /// Indicates that the implementing class will configure the host.
```
Comment (Scooletz) [inline src/NServiceBus.Hosting.Azure/Roles/IConfigureThisHost.cs]: I wonder whether or not should we mention the dynamic hosting here.

**Comment by Scooletz** [comment]
One comment only. All the rest :+1:

File: src/NServiceBus.Hosting.Azure/Roles/IConfigureThisHost.cs (hunk 1 above)
Comment (MarcinHoppe) [inline src/NServiceBus.Hosting.Azure/Roles/IConfigureThisHost.cs]: @Scooletz Slightly improved wording.

**Pull request merged**